pkg/dmaasbackup/nonperiodic_backup.go

```diff
@@ -121,44 +122,97 @@ func (d *dmaasBackup) cleanupNonPeriodicSchedule(dbkp *v1alpha1.DMaaSBackup) err
 		return nil
 	}
 
-	backupList, err := d.backupLister.List(
-		labels.SelectorFromSet(map[string]string{
-			v1alpha1.DMaaSBackupLabelKey:  dbkp.Name,
-			velerov1api.ScheduleNameLabel: schedule.ScheduleName,
-		}),
-	)
-
+	backupList, err := d.listScheduledBackups(dbkp, schedule.ScheduleName)
 	if err != nil {
 		return errors.Wrapf(err, "failed to get list of backup for schedule=%s", schedule.ScheduleName)
 	}
 
+	if len(backupList) < retainCount {
+		d.logger.Debugf("Number of backups are %v, required %v backups to trigger cleanup",
+			len(backupList),
+			retainCount)
+		return nil
+	}
+
 	defer d.logger.Debug("Completed cleanup for non-periodic schedule")
 
 	sort.Sort(sort.Reverse(byBackupCreationTimeStamp(backupList)))
```

Comment:
is that better to use backup creation time or backup completion time?

Reply:
Since velero run one backup at a time, We can use backup completion time also. But, With backup completion time, stalled backup will not get deleted until TTL gets expired.